integration-tests/google-pubsub/src/test/java/org/apache/camel/quarkus/component/google/pubsub/it/GooglePubsubTest.java

```diff
@@ -56,6 +74,220 @@ public void jacksonSerializer() {
+                .extract().asString(),
+                Matchers.is("6"));
+        LOG.info("Message \"6\" received.");
+
+    }
+
+    @Test
+    @Order(5)
+    public void testAck() throws Exception {
+        LOG.info("Start of the acking test");
+        //enable ack
+        RestAssured.given()
+                .body(false)
+                .post("/google-pubsub/setFail")
+                .then()
+                .statusCode(201);
+        //successful run
+        RestAssured.given()
+                .queryParam("toEndpoint", GooglePubSubRoutes.ACK_DIRECT_IN)
+                .body("1")
+                .post("/google-pubsub/sendToEndpoint")
+                .then()
+                .statusCode(201);
+        LOG.info("Message \"1\" was sent and should be Acked.");
+
+        await().atMost(5, TimeUnit.SECONDS).untilAsserted(() -> RestAssured.given()
+                .get("/google-pubsub/receive/mock/" + GooglePubSubRoutes.ACK_MOCK_RESULT)
+                .then()
+                .statusCode(200)
+                .body(Matchers.is("1")));
+        LOG.info("Message \"1\" was received.");
+        //once message is received, reset mock
+        RestAssured.given()
+                .get("/google-pubsub/resetMock/" + GooglePubSubRoutes.ACK_MOCK_RESULT)
+                .then()
+                .statusCode(201);
+        //failing run
+        //disable ack
+        RestAssured.given()
+                .body(true)
+                .post("/google-pubsub/setFail")
+                .then()
+                .statusCode(201);
+
+        RestAssured.given()
+                .queryParam("toEndpoint", GooglePubSubRoutes.ACK_DIRECT_IN)
+                .body("2")
+                .post("/google-pubsub/sendToEndpoint")
+                .then()
+                .statusCode(201);
+        LOG.info("Message \"2\" was sent and should be Nacked.");
+        //wait to be sure that the nacked message was not delivered
+        Thread.sleep(1000);
+
+        RestAssured.given()
+                .get("/google-pubsub/receive/mock/" + GooglePubSubRoutes.ACK_MOCK_RESULT)
+                .then()
+                .statusCode(200)
+                .body(Matchers.is(""));
+        LOG.info("Message \"2\" was not delivered.");
+
+        //enable ack
+        RestAssured.given()
+                .body(false)
+                .post("/google-pubsub/setFail")
+                .then()
+                .statusCode(201);
+        LOG.info("Acking was enabled.");
+
+        //assert redelivered message
+        await().atMost(20, TimeUnit.SECONDS).untilAsserted(() -> RestAssured.given()
+                .get("/google-pubsub/receive/mock/" + GooglePubSubRoutes.ACK_MOCK_RESULT)
+                .then()
+                .statusCode(200)
+                .body(Matchers.is("2")));
+
+        LOG.info("Message \"2\" was delivered after acking was enabled.");
+    }
+
+    @Test
+    @Order(6)
+    public void stopConsumerTest() throws Exception {
+        RestAssured.given()
```

Comment:
I added a comment to the follow-up issue to answer you (https://github.com/apache/camel-quarkus/issues/3957#issuecomment-1217698290)

>The problem seems to be somehow connected to the close of the consumer in the native mode.There is a test stopConsumerTest, which is closing the consumer after all the tests (and requires method order to achieve this)

I would keep the method here, because it forces the right closure of the consumer in the native mode. Once https://github.com/apache/camel-quarkus/issues/3957 is solved, this method should be removed.

Reply:
I commented the code of the test method and pointed to the issue https://github.com/apache/camel-quarkus/issues/3957

Reply:
Ok, great many thanks for sharing :+1: